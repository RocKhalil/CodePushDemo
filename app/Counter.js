
import React from "react";
import { View, Button } from "react-native";
export default class Counter extends React.Component {
  state = {
    count: 0
  };
  increment = () => {
    this.setState(prevState => ({ count: prevState.count + 1 }));
  };
  render() {
    return (
      <Button title={`Count ${this.state.count}`} onPress={this.increment} />
    );
  }
}