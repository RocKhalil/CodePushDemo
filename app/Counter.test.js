import React from "react";
import Counter from "./Counter";
import renderer from "react-test-renderer";
import { Button } from "react-native";

it("counter", () => {
  const rendered = renderer.create(<Counter />);
  const counter = rendered.getInstance();
  expect(counter.state.count).toBe(0);
  const button = rendered.root.findByType(Button).props.onPress();
  expect(counter.state.count).toBe(1);
});

it("counter snapshot with initial state", () => {
  const rendered = renderer.create(<Counter />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("counter snapshot with one Press", () => {
  const rendered = renderer.create(<Counter />);
  const button = rendered.root.findByType(Button).props.onPress();
  expect(rendered.toJSON()).toMatchSnapshot();
});