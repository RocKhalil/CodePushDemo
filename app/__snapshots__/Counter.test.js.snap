// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter snapshot with initial state 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={undefined}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Count 0
    </Text>
  </View>
</View>
`;

exports[`counter snapshot with one Press 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={undefined}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Count 1
    </Text>
  </View>
</View>
`;
